### PR TITLE
Add Go CLI to OpenWhisk - Ansible

### PR DESCRIPTION
**Pull request by dubeejw**
- Setup Ansible so that Go CLI can be published with nginx (not included)

### CODE REVIEW

**Comment by dubeejw** [comment]
@mdeuser, please review.

**Comment by mdeuser** [comment]
@dubeejw - I can take a look, but we should have one of the ansible experts eyes on this as well.

**Comment by dubeejw** [comment]
@domdom82, can you take a look at this?

File: ansible/environments/mac/group_vars/all [hunk 1]
```diff
@@ -3,6 +3,7 @@ nginx_conf_dir: /Users/Shared/nginx
 consul_conf_dir: /Users/Shared/consul
 whisk_logs_dir: /Users/Shared/wsklogs
 cli_conf_dir: /Users/Shared/cli
+go_cli_conf_dir: /tmp/go-cli
```
Comment (lehoanganh) [inline ansible/environments/mac/group_vars/all]: pls place it under shared folder `/Users/Shared` like other folders. For mac, it's diffent.

File: ansible/roles/go-cli/tasks/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+- name: "Ensure go-cli directory exists"
+  file:
+    path: "{{ go_cli_conf_dir }}"
+    state: directory
+
+- name: "Pull the {{ docker_image_tag }} image of go-cli"
+  shell: "docker pull {{ docker_registry }}whisk/go-cli:{{ docker_image_tag }}"
+  when: docker_registry != ""
+
+# TODO: Publish compressed binaries using nginx
```
Comment (lehoanganh) [inline ansible/roles/go-cli/tasks/deploy.yml]: you don't want to publish the tarball here?

File: ansible/roles/go-cli/templates/default.props.j2 [hunk 3]
```diff
@@ -0,0 +1,2 @@
+CLI_API_HOST={{ cli_api_host | default(groups['edge']) | first }}
```
Comment (lehoanganh) [inline ansible/roles/go-cli/templates/default.props.j2]: the go-cli needs also `default.props`?

File: ansible/roles/go-cli/tasks/deploy.yml (hunk 2 above)
Comment (dubeejw) [inline ansible/roles/go-cli/tasks/deploy.yml]: Why not? The tarballs will be in the Docker container at this point.

Comment (dubeejw) [inline ansible/roles/go-cli/tasks/deploy.yml]: @lehoanganh, at this point in the Ansible file for the Python CLI, a generated tarball is moved to the nginx config directory. This is what we want to do here.

Comment (dubeejw) [inline ansible/roles/go-cli/tasks/deploy.yml]: @lehoanganh, we have @houshengbo working on the publishing aspect mentioned here.

Comment (lehoanganh) [inline ansible/roles/go-cli/tasks/deploy.yml]: I try to understand the process. The `go-cli` would do the same process like python cli does?

If so, at this comment

```
# TODO: Publish compressed binaries using nginx
```

Why don't we have steps to get the tarball in docker container and place it somewhere under nginx?

Comment (dubeejw) [inline ansible/roles/go-cli/tasks/deploy.yml]: Vincent will be working on that part. He'll add it in afterward.

Comment (dubeejw) [inline ansible/roles/go-cli/tasks/deploy.yml]: Plus the process isn't as simple because we have numerous tar/zip files now.

Comment (dubeejw) [inline ansible/roles/go-cli/tasks/deploy.yml]: See https://github.com/openwhisk/openwhisk/issues/577

**Pull request merged**